media/js/base/mozilla-fxa-form.js

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// create namespace
+if (typeof Mozilla === 'undefined') {
+    var Mozilla = {};
+}
+
+Mozilla.FxaForm = (function(Mozilla) {
+    'use strict';
+
+    var fxaForm = document.getElementById('fxa-email-form');
+    var fxaSubmitButton = document.getElementById('fxa-email-form-submit');
+
+    // swap form action for Fx China re-pack
+    function init() {
+        // disable form while we check distribution
+        fxaSubmitButton.disabled = true;
+
+        var mozillaonlineAction = fxaForm.dataset.mozillaonlineAction;
+
+        if (mozillaonlineAction) {
+            Mozilla.Client.getFirefoxDetails(function(data) {
```

Comment:
Hm, I do think the current assumption is that this form can/will only be displayed to Fx users. (This same assumption is made on the iframe'd form.)

If this assumption is incorrect, we can file a follow-up PR to alter some of this JS.

Reply:
I have to embed it on /download/thanks and users of all types end up on that page. I'm only *displaying* it to Firefox users though so maybe there's something I can do to avoid the JS error if the form isn't displayed.